iohub/ngff.py

```diff
@@ -396,6 +396,39 @@ def _storage_options(self):
     def _member_names(self):
         return self.array_keys()
 
+    @property
+    def data(self):
+        """.. Warning:
+            This property does *NOT* aim to retrieve all the arrays.
+            And it may also fail to retrive any data if arrays exist but
+            are not named conventionally.
+
+        Alias for an array named '0' in the position,
+        which is usually the raw data (or the finest resolution in a pyramid).
+
+        Returns
+        -------
+        ImageArray
+
+        Raises
+        ------
+        KeyError
+            If no array is named '0'.
+
+        Notes
+        -----
+        Do not depend on this in non-interactive code!
+        The name is hard-coded and is not guaranteed
+        by the OME-NGFF specification.
+        """
+        try:
+            return self["0"]
```

Comment:
Could `self["0"]` become `self.array_keys()[0]`? 

Instead of failing if "0" isn't present, this would return the first entry in the `sorted` list of `array_keys()`. This suggestion could lead to confusing behavior if `self.data` returns a different result when you add a new position with a earlier sorted name, but I think I'd prefer this confusing behavior to the confusion of `self.data` returning `KeyError` when I know that there's data in the store.

Reply:
My feeling is that we should be very explicit and deterministic about what data the software is reading. IMO the first item of a 'sorted' list is still almost random if the user did not take a look at what files (arrays) they actually have in the folder (group), and if they did, a file name (array key) should be used instead of this shorthand. In short, this attribute should not be used when there *might* be anything other than array `'0'` in the position, and that is (hopefully) emphasized multiple times in the documentation. And I think if `KeyError` is confusing for you, we should aim to improve the error message.

Reply:
Makes total sense. Thanks 🙏

Reply:
> IMO the first item of a 'sorted' list is still almost random

An example would be a position with arrays `['raw', 'projection', 'mask']`, if the list is sorted alphabetically the first item would be 'mask', and is almost certainly not expected, but if they have the same size and dtype this cannot be discovered automatically. This kind of behavior can result in a cryptic errors (better scenario) or false analysis that leaks all the way into a published work (worst case).